website/src/components/Edit/SequencesForm.tsx

```diff
@@ -1,117 +1,178 @@
 import { type Dispatch, type FC, type SetStateAction } from 'react';
 
-import { type KeyValuePair, type Row } from './InputField';
-import { mapErrorsAndWarnings, type SequenceEntryToEdit } from '../../types/backend.ts';
+import { type SequenceEntryToEdit } from '../../types/backend.ts';
+import type { ReferenceGenomesLightweightSchema } from '../../types/referencesGenomes.ts';
 import { FileUploadComponent } from '../Submission/FileUpload/FileUploadComponent.tsx';
 import { PLAIN_SEGMENT_KIND, VirtualFile } from '../Submission/FileUpload/fileProcessing.ts';
 
-function generateAndDownloadFastaFile(
-    accessionVersion: string,
-    sequenceData: string,
-    segmentKey?: string,
-    isSingleSegment: boolean = false,
-) {
-    const fileName =
-        isSingleSegment || !segmentKey ? `${accessionVersion}.fasta` : `${accessionVersion}_${segmentKey}.fasta`;
-
-    const header = isSingleSegment || !segmentKey ? accessionVersion : `${accessionVersion}_${segmentKey}`;
-
-    const fileContent = `>${header}\n${sequenceData}`;
+function generateAndDownloadFastaFile(fastaHeader: string | null, sequenceData: string | null) {
+    let fileContent = '';
+    let trimmedHeader = '';
+
+    if (fastaHeader === null && sequenceData === null) {
+        fileContent = '';
+    } else {
+        if (fastaHeader === null) {
+            throw new Error(
+                'Internal Error: sequenceData exists but fastaHeader is empty - contact your administrator',
+            );
+        }
+
+        trimmedHeader = fastaHeader.replace(/\s+/g, '');
+        fileContent = `>${trimmedHeader}\n${sequenceData ?? ''}`;
+    }
 
     const blob = new Blob([fileContent], { type: 'text/plain' });
     const url = URL.createObjectURL(blob);
 
     const a = document.createElement('a');
     a.href = url;
-    a.download = fileName;
+    a.download = `${trimmedHeader || 'sequence'}.fasta`;
     a.click();
 
     URL.revokeObjectURL(url);
 }
 
+type EditableSequenceFile = {
+    key: string;
+    label: string;
+    fastaHeader: string | null;
+    value: string | null;
+    initialValue: string | null;
+};
+
 export class EditableSequences {
```

Comment:
The counts are off:

<img width="776" height="221" alt="Image" src="https://github.com/user-attachments/assets/1953ccf7-5359-4dea-8770-af9573df20ca" />

Doesn't really matter, but this probably increases the `nextKey` once too often.